### PR TITLE
Fix nodejs breakage and re-add bear tests

### PR DESCRIPTION
**Pull request by jayvdb**

### CODE REVIEW

**Comment by jayvdb** [comment]
Amend or cherry-pick, as useful.

File: Dockerfile [hunk 1]
```diff
@@ -135,14 +148,12 @@ RUN luarocks install luacheck
...
 RUN npm install -g \
-  alex \
   autoprefixer \
   bootlint \
   coffeelint \
   complexity-report \
   csslint \
   dockerfile_lint \
-  eslint \
   jshint \
```
Comment (jayvdb) [inline Dockerfile]: Should also remove jshint, if it is installed from openSUSE pkg

File: Dockerfile [hunk 2]
```diff
@@ -135,14 +148,12 @@ RUN luarocks install luacheck
 # NPM setup
 # FIXME: we should use package.json from coala
 RUN npm install -g \
-  alex \
```
Comment (jayvdb) [inline Dockerfile:138]: I dont know why alex doeent pass. Actually only one of many tests fails :/

Comment (AbdealiLoKo) [reply Dockerfile:138]: @jayvdb Are you aware of the conflicting package name in the haskell compiler (ghc i think) called alex ? We've had issues before where the binary used is the wrong binary ...

**Comment by AbdealiLoKo** [comment]
Question: Can we just use cib ? Is it not yet stable enough ?
CC @Adrianzatreanu

File: Dockerfile (hunk 2 above)
Comment (jayvdb) [reply Dockerfile:138]: Ya, I did see that, but it is only `AlexBearTest.test_invalid_files` that fails, and I doubt that `alex` is installed on this rather empty openSUSE image.  And https://github.com/coala/coala-bears/commit/fa70e924 added checks for this , and those tests pass.
A neater solution would be to disable only the one failing test.

**Comment by jayvdb** [comment]
Circle node is v5.5.0 ; maybe we need to set node to v5 on openSUSE.

File: Dockerfile [hunk 3]
```diff
@@ -133,23 +137,10 @@ RUN julia -e 'Pkg.add("Lint")'
...
-  postcss-cli \
-  remark-cli \
-  tslint \
-  typescript \
-  ramllint \
-  write-good
+# Extract dependencies from coala-bear package.json
+# typescript is a peer dependency
+RUN npm install -g typescript \
+    $(sed -ne '/~/{s/^[^"]*"//;s/".*"~/@/;s/",*//;p}' coala-bears/package.json)
```
Comment (underyx) [inline Dockerfile:143]: 😳 

How reliable is this? Couldn't we just do

```
WORKDIR /coala-bears/package.json
RUN npm install --global
```

or something similar?

**Comment by Mixih** [comment]
ack 
0e44724
000859c
785538c
3584d5d

**Comment by underyx** [comment]
unack https://github.com/coala/docker-coala-base/pull/44/commits/000859cd0c3e85316ba932813d2f9f4bb3ee0982 until my comment is answered

feel free to reack it if you've seen my comment and think it's okay, @jayvdb

**Review by underyx — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 3 above)
Comment (jayvdb) [reply Dockerfile:143]: No.  That installs `coala-bears` globally, and none of the dependencies are global.

There are a few alternatives, all ugly.

> How reliable is this?

We now have bear testing again, so we can see if it breaks.
See bottom of [log](https://api.travis-ci.org/jobs/179452542/log.txt?deansi=true)

**Comment by jayvdb** [comment]
reack 000859c

**Review by underyx — APPROVED** [review]

**Pull request merged**